processor/src/main/java/org/mapstruct/ap/internal/model/NestedTargetPropertyMappingHolder.java

```diff
@@ -361,8 +365,19 @@ private GroupedTargetReferences groupByTargetReferences( ) {
             Map<String, Set<MappingReference>> singleTargetReferences = new LinkedHashMap<>();
             for ( MappingReference mapping : mappingReferences.getMappingReferences() )  {
                 TargetReference targetReference = mapping.getTargetReference();
+                SourceReference sourceReference = mapping.getSourceReference();
                 String property = first( targetReference.getPropertyEntries() );
                 MappingReference newMapping = mapping.popTargetReference();
+
+                if ( newMapping == null && method.getSourceParameters().size() == 1 && sourceReference != null && sourceReference.getParameter() != null &&
+                    sourceReference.getPropertyEntries().isEmpty() &&
+                    !targetReference.getPropertyEntries().isEmpty() ) {
+                    List<String> newPathProperties = new ArrayList<>( targetReference.getPathProperties() );
+                    newPathProperties.add( targetReference.getPropertyEntries().get( 0 ) );
+                    TargetReference newTargetReference = new TargetReference( null, Collections.singletonList( "." ),  newPathProperties);
+                    newMapping = new MappingReference( mapping.getMapping(), newTargetReference, sourceReference );
```

Comment:
So, what you are doing here is creating a new "." mapping when you only have a source parameter (no properties) in the mapping.. But then I get lost...